pkg/workloadmeta/types.go

```diff
@@ -804,6 +820,57 @@ func printHistory(out io.Writer, history v1.History) {
 
 var _ Entity = &ContainerImageMetadata{}
 
+type Process struct {
+	EntityID // EntityID is the PID for now
+	EntityMeta
+
+	NsPid        int
+	ContainerId  string
+	CreationTime time.Time
+	Language     *string
```

Comment:
What do you think about using the `Language` type defined in `pkg/languagedetection` library so we keep them in sync?

https://github.com/DataDog/datadog-agent/blob/c970613506ac249c5148319e5a70817580c188c8/pkg/languagedetection/detector.go#L26-L28

Reply:
Indeed, we can use that. I was looking for the exact type and used `*string` as a place holder.

Reply:
why does `languagedetection` import workload meta ?

Reply:
@just-chillin moved the language definition to a subpackage in https://github.com/DataDog/datadog-agent/pull/17448 so once it's merged we can update it.

For now, it's ok to keep it as a string to avoid having one PR blocking the other.